### PR TITLE
unit tests for some components

### PR DESCRIPTION
**Pull request by mdegani**
Basic unit tests for: button, alert, counter, form, modal, navigator, logo.

### CODE REVIEW

**Comment by mdegani** [comment]
@SethDavenport Brendan is writing tests for the remaining components on a separate branch.

File: src/components/alert/index.test.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        expect(query).toBeTruthy();
+        expect(query.componentInstance).toBeTruthy();
+      });
+  }));
+});
+
+@Component({
+  selector: 'test',
+  template: `
+    <rio-alert></rio-alert>
```
Comment (SethDavenport) [inline src/components/alert/index.test.ts]: I'm not sure about this pattern.  Are you creating a fake parent component in order to instantiate the alert component under test?

It's more usual to use `builder.createAsyc` to instantiate the component under test directly and then do expectations on it. See here for an example: https://github.com/rangle/angular2-unit-test-app/blob/master/src/components/user-list/user-list.component.spec.ts#L46

`builder.createAsync(RioAlert)`

File: src/components/alert/index.test.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+describe('Component: Alert', () => {
+  let builder: TestComponentBuilder;
+
+  beforeEachProviders(() => [RioAlert]);
+  beforeEach(inject([TestComponentBuilder],
+    function (tcb: TestComponentBuilder) {
+      builder = tcb;
+    }));
+
+  it('should inject the component', inject([RioAlert],
```
Comment (SethDavenport) [inline src/components/alert/index.test.ts]: I find this sort of 'it should get created' style tests to have limited value in practice.

File: src/components/alert/index.test.ts (hunk 1 above)
Comment (SethDavenport) [inline src/components/alert/index.test.ts]: It would also be good to test that the generated DOM gets the right CSS classes based on the value of `type`.

Comment (mdegani) [inline src/components/alert/index.test.ts]: I'll make that change.

Comment (mdegani) [inline src/components/alert/index.test.ts]: You're right Seth.  I'm going to be gradually converting the tests over this morning.  I got the pattern from the tests that Angular 2 CLI automatically generates.  Maybe it makes sense to test boilerplate, but should be changed once behaviour is introduced.

File: src/components/alert/index.test.ts (hunk 2 above)
Comment (mdegani) [inline src/components/alert/index.test.ts]: Yes, I agree.  Fixing today.

File: src/components/alert/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+import { RioAlert } from './index';
+
+describe('Component: Alert', () => {
+  // let builder: TestComponentBuilder;
+
+  beforeEachProviders(() => [RioAlert]);
+  // beforeEach(inject([TestComponentBuilder],
+  //   function (tcb: TestComponentBuilder) {
+  //     builder = tcb;
+  //   }));
```
Comment (SethDavenport) [inline src/components/alert/index.test.ts]: Commented-out code!

Comment (mdegani) [inline src/components/alert/index.test.ts]: Thanks.  There is still a lot to clean up on this branch.

**Pull request closed without merging**